### PR TITLE
cross-x86_64-w64-mingw32: update to 8.0.0

### PR DESCRIPTION
**Pull request by alekseyt**
* Mingw-w64 8.0.0
* GCC 10.2.0
* Other dependencies harmonized with mainline
* Incorporates #16891

### CODE REVIEW

**Comment by ericonr** [comment]
Know any simple project I can easily test the resulting toolchain with?

**Comment by alekseyt** [comment]
Hmm. Maybe Lua: https://www.lua.org/versions.html#5.4

```bash
$ make mingw CC=x86_64-w64-ming32-gcc
```

and then run it with Wine: `wine ./src/lua.exe`

This is C. As for C++, perhaps CppUnit will do? https://www.freedesktop.org/wiki/Software/cppunit/

```bash
$ LDFLAGS="-static-libgcc -static-libstdc++ -static" ./configure --host=x86_64-w64-mingw32 --disable-doxygen
$ make
```

and then again run with wine: `WINEPATH=/usr/x86_64-w64-mingw32/bin wine ./examples/simple/simple.exe`

Generally any simple project should be buildable with regard to cross-compiling specifics.

Please let me know if i can help with anything else.

File: srcpkgs/cross-x86_64-w64-mingw32/template [hunk 1]
```diff
@@ -1,17 +1,17 @@
 # Template file for 'cross-x86_64-w64-mingw32'
-_gcc_version=9.3.0
-_binutils_version=2.32
-_gmp_version=6.1.2
-_mpfr_version=4.0.1
+_gcc_version=10.2.0
+_binutils_version=2.34
+_gmp_version=6.2.0
+_mpfr_version=4.0.2
```
Comment (ericonr) [inline srcpkgs/cross-x86_64-w64-mingw32/template:5]: The gcc 10.2 PR is using mpfr 4.1.0, from what I can see. Should we perhaps use that instead?

Comment (alekseyt) [reply srcpkgs/cross-x86_64-w64-mingw32/template]: 69e1505

**Comment by alekseyt** [comment]
I've also noticed that `libstdc++fs.a` is being stripped, so i've added it to `nostrip_files`.

**Comment by ericonr** [comment]
I've also found out that musl's wine is misbehaving, since it craps out when exiting wine.

Thanks for the examples, I'm testing with the change to mpfr.

**Comment by ericonr** [comment]
Thanks!

**Pull request merged**